tywrap_ir/tywrap_ir/ir.py

```
@@ -470,9 +505,148 @@ def _extract_function(
         is_generator=is_generator,
         type_params=type_params,
+        method_kind=method_kind,
+        overloads=_extract_overloads(obj),
     )
 
 
+# Guard so the 3.10 degradation warning is emitted at most once per process.
+_overload_warning_emitted = False
+
+
+def _extract_overloads(obj: Any) -> List[IROverload]:
```

Ruff (0.15.14)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `obj`

(ANN401)